src/server/search/search_accessor.h

```diff
@@ -0,0 +1,167 @@
+// Copyright 2023, DragonflyDB authors.  All rights reserved.
+// See LICENSE for licensing terms.
+//
+
+#pragma once
+
+#include <absl/container/flat_hash_map.h>
+
+#include <jsoncons/json.hpp>
+
+#include "core/json_object.h"
+#include "core/search/search.h"
+#include "server/container_utils.h"
+
+extern "C" {
+#include "redis/listpack.h"
+#include "redis/object.h"
+};
+
+namespace dfly {
+
+using namespace std;
+
+namespace {
+
+string_view SdsToSafeSv(sds str) {
+  return str != nullptr ? string_view{str, sdslen(str)} : ""sv;
+}
+
+}  // namespace
+
+// Document accessors allow different types (json/hset) to be hidden
+// behind a document interface for quering fields and serializing.
+struct BaseAccessor : public search::DocumentAccessor {
+  using FieldConsumer = search::DocumentAccessor::FieldConsumer;
+
+  virtual SearchDocData Serialize() const = 0;
```

Comment:
remind what `Serialize` is used for?

Reply:
Convert underlying type to a map<string, string> to be sent as a reply, added a comment